# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/quota/QuotaManager.h
# Commit: 6180f2e753bd
# Full Hash: 6180f2e753bd2efb4905c6a70eb6607144807d5b
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2024-10-11 10:49:37
# Regressor Bug: 1866402
# File Overlap Count: 1
# Description:
#   Bug 1866402 - Add support for initializing corresponding origin to QuotaManager::OpenClientDirectory; r=devtools-reviewers,nchevobbe
#   
#   QuotaManager::OpenClientDirectory can now get initialization status of
#   corresponding origin and initialize the origin if it's needed.
#   
# ==============================================================================

diff -r f25dfc157ef7 -r 6180f2e753bd dom/quota/QuotaManager.h
--- a/dom/quota/QuotaManager.h	Fri Oct 11 00:06:08 2024 +0000
+++ b/dom/quota/QuotaManager.h	Fri Oct 11 00:06:09 2024 +0000
@@ -295,7 +295,7 @@
   // must be always done explicitly before the lock object is destroyed (when
   // the last strong reference is removed).
   RefPtr<ClientDirectoryLockPromise> OpenClientDirectory(
-      const ClientMetadata& aClientMetadata,
+      const ClientMetadata& aClientMetadata, bool aCreateIfNonExistent = true,
       Maybe<RefPtr<ClientDirectoryLock>&> aPendingDirectoryLockOut = Nothing());
 
   RefPtr<ClientDirectoryLock> CreateDirectoryLock(